Sources/BraintreeApplePay/BTApplePayClient.swift

```diff
@@ -80,26 +82,29 @@ import BraintreeCore
             self.apiClient.post("v1/payment_methods/apple_payment_tokens", parameters: parameters) { body, _, error in
                 if let error = error as NSError? {
                     if error.code == BTCoreConstants.networkConnectionLostCode {
-                        self.apiClient.sendAnalyticsEvent("ios.apple-pay.network-connection.failure")
+                        self.apiClient.sendAnalyticsEvent("apple-pay:tokenize:network-connection:failed")
+                    } else {
+                        self.apiClient.sendAnalyticsEvent("apple-pay:tokenize:failed")
```

Comment:
Thinking through this again, if we want to calculate conversion failure rates, we need to do:
```
total_failed_events / (total_start_events + total_failed_events) X 100
```

We know that our start event is `apple-pay:payment-request:started` (simple, since only triggered once). I think if we keep the code as-is, then we'll need to add up both `# of apple-pay:tokenize:failed` + `# of apple-pay:tokenize:network-connection:failed` to get `total_failed_events`.

So maybe we _do_ want to send `apple-pay:tokenize:network-connection:failed` alongside a `apple-pay:tokenize:failed`? We can treat `failed` as an umbrella for all failures. If we do this route - what we need to be sure of, is just documenting the behavior of the all-encompassing failed event, and making sure it is not fired more than once per error route.

Does that make sense?

Whatever we do, it should be consistent & documented so we know exactly how to calculate conversion later on.

Reply:
I actually think this was more in-line with our plan for [PPCP events](https://paypal.sharepoint.com/:x:/r/sites/BTSDK/_layouts/15/Doc.aspx?sourcedoc=%7BAAA7C353-5744-4F21-AC7D-DF85D8605817%7D&file=Northstar%20Analytics%20Events.xlsx&action=default&mobileredirect=true&cid=1b0afccd-68ab-448f-9760-60ed14df9811). Have a single/high-level "failed" event that gets fired for all failures, for the purpose of calculating conversion.

Reply:
I think that makes a ton of sense and is a great idea! Then we can calculate conversion for all failures (including network connection failures). That will also allow us to calculate the network connection failure rate as a percentage of the total failures, which I think would likely be more useful as well so we can really drill into that data alongside the success/failure data.